staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/config/authentication.go

```diff
@@ -102,7 +107,7 @@ func (c *defaultAuthenticationInfoResolver) clientConfig(target string) (*rest.C
 	}
 
 	// if we're trying to hit the kube-apiserver and there wasn't an explicit config, use the in-cluster config
-	if target == "kubernetes.default.svc" {
+	if target == "kubernetes.default.svc" && !c.skipInCluster {
```

Comment:
This rubs me the wrong way.  Modifying my code for production smells wrong.  What about performing a check and doing something different if inclusterconfigs are available in the test?